sdk/src/main/scala/com/horizen/SidechainApp.scala

```diff
@@ -248,6 +261,8 @@ class SidechainApp @Inject()
       SidechainWalletCswDataProviderCSWDisabled()
   }
 
+  //Waits for actors to start
+  Thread.sleep(500)
```

Comment:
this part is not clear to me.
why do we need to sleep?
how it worked before?

Reply:
I have moved to the base class some operations that were before performed between the creation of the CertificateSubmitter actor and the publishing of SidechainApplicationStart event. So it happens that the CertificateSubmitter actor is not able to subscribe to the SidechainApplicationStart event in time to receive it and this causes that the CertificateSubmitter never starts. I have tried to see if there were ways to know if an actor has started or not, but I haven't found anything so I have put a sleep. Probably I can reduce the sleep time, I was very conservative. Same for AccountSidechainApp.

Reply:
let's check deeply the behavior